src/frequenz/dispatch/_dispatcher.py

```diff
@@ -214,15 +217,37 @@ def __init__(
         Args:
             microgrid_id: The microgrid id.
             server_url: The URL of the dispatch service.
-            key: The key to access the service.
+            key: The key to access the service, deprecated, use `auth_key` instead.
+            auth_key: The authentication key to access the service.
+            sign_secret: The secret to sign the requests, optional
             call_timeout: The timeout for API calls.
             stream_timeout: The timeout for streaming API calls.
+
+        Raises:
+            ValueError: If both or neither `key` and `auth_key` are provided
         """
         super().__init__()
 
+        if key is not None and auth_key is not None:
+            raise ValueError(
+                "Both 'key' and 'auth_key' are provided, please use only 'auth_key'."
+            )
+
+        if key is None and auth_key is None:
+            raise ValueError(
+                "Either 'key' or 'auth_key' must be provided to access the dispatch service."
```

Comment:
Niptpick: Since `key` is deprecated I wouldn't mention it here.
```suggestion
                "'auth_key' must be provided to access the dispatch service."
```